analyzer/src/main/kotlin/managers/Pub.kt

```diff
@@ -299,32 +305,31 @@ class Pub(
         return packageReferences.toSortedSet()
     }
 
-    private val analyzerResultCacheAndroid = mutableMapOf<String, ProjectAnalyzerResult>()
+    private val analyzerResultCacheAndroid = mutableMapOf<String, List<ProjectAnalyzerResult>>()
 
-    private fun scanAndroidPackages(packageInfo: JsonNode): ProjectAnalyzerResult? {
+    private fun scanAndroidPackages(packageInfo: JsonNode): List<ProjectAnalyzerResult> {
         val packageName = packageInfo["description"]["name"].textValueOrEmpty()
 
         // We cannot find packages without a valid name.
-        if (packageName.isEmpty()) return null
+        if (packageName.isEmpty()) return emptyList()
 
-        val projectRoot = reader.findProjectRoot(packageInfo) ?: return null
+        val projectRoot = reader.findProjectRoot(packageInfo) ?: return emptyList()
         val androidDir = File(projectRoot, "android")
         val packageFile = File(androidDir, "build.gradle")
 
         // Check for build.gradle failed, no Gradle scan required.
-        if (!packageFile.isFile) return null
+        if (!packageFile.isFile) return emptyList()
 
         log.info { "Analyzing Android dependencies for package '$packageName' using Gradle version $GRADLE_VERSION." }
 
         return if (analyzerResultCacheAndroid.containsKey(packageName)) {
-            analyzerResultCacheAndroid[packageName]
+            analyzerResultCacheAndroid.getValue(packageName)
         } else {
             // Use the latest 5.x Gradle version as Flutter / its Android Gradle plugin does not support Gradle 6 yet.
             Gradle("Gradle", androidDir, analyzerConfig, repoConfig, GRADLE_VERSION)
-                .resolveDependencies(listOf(packageFile))[packageFile]
-                ?.also {
-                    analyzerResultCacheAndroid[packageName] = it
-                }
+                .resolveDependencies(listOf(packageFile))
+                .getValue(packageFile)
+                .also { analyzerResultCacheAndroid[packageName] = it }
```

Comment:
We could use the chance and rewrite the whole block as

        return analyzerResultCacheAndroid.getOrPut(packageName) {
            // Use the latest 5.x Gradle version as Flutter / its Android Gradle plugin does not support Gradle 6 yet.
            Gradle("Gradle", androidDir, analyzerConfig, repoConfig, GRADLE_VERSION)
                .resolveDependencies(listOf(packageFile))
                .getValue(packageFile)
        }

Reply:
Done.